File: backend/registration-service/app/com/radovan/play/utils/NatsUtils.java
package com.radovan.play.utils;

import io.nats.client.Connection;
import io.nats.client.Nats;
import jakarta.annotation.PostConstruct;

public class NatsUtils {

    private Connection nc;

    @PostConstruct
    public void init() {
        try {
            this.nc = Nats.connect("nats://localhost:4222");
            System.out.println("*** NATS connection has been established!");
        } catch (Exception e) {
            System.err.println("*** Error accessing NATS server!");
            e.printStackTrace();
        }
    }

    public Connection getConnection() {
        return nc;
    }


    public void closeConnection() {
        try {
            if (nc != null) {
                nc.close();
                System.out.println("*** NATS connection closed!");
            }
        } catch (Exception e) {
            System.err.println("*** Error closing NATS connection!");
            e.printStackTrace();
        }
    }
}

File: backend/infraction-service/app/com/radovan/play/utils/NatsUtils.scala
package com.radovan.play.utils

import io.nats.client.{Connection, Nats}
import jakarta.annotation.{PostConstruct, PreDestroy}

class NatsUtils  {

  private var nc: Connection = _

  @PostConstruct
  def init(): Unit = {
    try {
      nc = Nats.connect("nats://localhost:4222")
      println("*** NATS connection has been established!")
    } catch {
      case e: Exception =>
        System.err.println("*** Error accessing NATS server!")
        e.printStackTrace()
    }
  }

  def getConnection: Connection = nc

  @PreDestroy
  def closeConnection(): Unit = {
    try {
      if (nc != null) {
        nc.close()
        println("*** NATS connection closed!")
      }
    } catch {
      case e: Exception =>
        System.err.println("*** Error closing NATS connection!")
        e.printStackTrace()
    }
  }
}


